purchase_order_line_stock_move_cancel/models/purchase_order_line.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+from odoo.fields import Command
+
+
+class PurchaseOrderLine(models.Model):
+
+    _inherit = "purchase.order.line"
+
+    can_cancel_moves = fields.Boolean(
+        compute="_compute_can_cancel_moves",
+        help="This is a technical fields in order to display the cancel button",
+    )
+
+    def _compute_can_cancel_moves(self):
+        """
+        All moves from a purchase order line should not be cancelled or done
+        """
+        for line in self:
+            line.can_cancel_moves = bool(
+                line.move_ids
+                and all(move.state not in ("done", "cancel") for move in line.move_ids)
```

Comment:
@luisDIXMIT This might happen because we should use any, not all

Reply:
@rousseldenis Can you review?

Reply:
Yes I will try tomorrow. On other things right now.